common/src/main/java/apoc/export/cypher/formatter/CypherFormatterUtils.java

```diff
@@ -157,6 +159,18 @@ public static boolean isUniqueLabelFound(Node node, Map<String, Set<String>> uni
         }
     }
 
+    /*
+     * Returns true if there exists no other relationship with the same start node, end node and type
+     */
+    public static boolean isUniqueRelationship(Relationship rel) {
+        return StreamSupport.stream(
+                        rel.getStartNode()
+                                .getRelationships(Direction.OUTGOING, rel.getType())
+                                .spliterator(),
+                        false)
+                .noneMatch(r -> !r.equals(rel) && r.getEndNode().equals(rel.getEndNode()));
+    }
```

Comment:
Does this really do what it says it does?

I'm thinking about this situation:

```cypher
create (a), (b)
with a,b 
create (a)-[:R]->(b)
with a,b
create (a)-[:R]->(b)
```

Relationship.equals looks like it's implemented based on id:

```java
@Override
public boolean equals(Object o) {
    return o instanceof Relationship && this.getId() == ((Relationship) o).getId();
}
```

Reply:
Ahh, never mind, now I see it.